CRM/ManualDirectDebit/Hook/QueryObjects/Contribution.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * Class CRM_ManualDirectDebit_Hook_QueryObjects_Contribution.
+ */
+class CRM_ManualDirectDebit_Hook_QueryObjects_Contribution extends CRM_Contact_BAO_Query_Interface {
+
+  /**
+   * Obtains list of fields for the query.
+   */
+  public function &getFields() {
+    $fields = [];
+
+    return $fields;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public function from($fieldName, $mode, $side) {
+    if (CRM_Utils_System::currentPath() != 'civicrm/contribute/search') {
```

Comment:
this is repeated here and below, can be moved to prive method maybe `isContributionSearchForm`